src/loader/__main__.py

```diff
@@ -1,12 +1,35 @@
 """Entry point for the loader."""
 
-from httpx import Client
+import logging
+
+from httpx import Client, HTTPError, Request, Response
 from letsbuilda.pypi import PyPIServices
+from logging_config import configure_logger
 
 from .loader import main
 
+configure_logger()
+logger = logging.getLogger(__name__)
+
+
+def log_request(request: Request) -> None:
+    """Log outgoing requests."""
+    logger.debug("%s %s - Waiting for response", request.method, request.url)
+
+
+def log_response(response: Response) -> None:
+    """Log request responses."""
+    request = response.request
+    logger.debug("%s %s - Status %s", request.method, request.url, response.status_code)
+
+    try:
+        response.raise_for_status()
+    except HTTPError:
+        logger.exception("Response had a non 2xx status code")
```

Comment:
I like the idea of the log level being based on the status code all in one log message. But I don't know if I like setting the `level = logger.debug` stuff that much, though. perhaps the logging library has a way to call a generic log method and pass in the level as a parameter? That would be much cleaner.

I just want to avoid the case where:
```
<blah blah>
GET /SOME URL - Status 500
<bunch of random logs>
EXCEPTION: Response had a non 2xx status code
<more blah blah>
```
wherein you need to go looking for _which_ response had a non 2xx status code.

```suggestion
    try:
        response.raise_for_status()
        level = logger.debug
    except HTTPError:
        level = logger.exception
    finally:
        level("%s %s - Status %s", request.method, request.url, response.status_code)
```